.github/generate-dist-files-size-diff.mjs

```diff
@@ -0,0 +1,118 @@
+  node .github/generate-dist-files-size-diff.mjs
+```
+ */
+
+if (!process.env.BASE_DIST_FILES) {
+    throw new Error('Missing or invalid "BASE_DIST_FILES" env variable.');
+}
+
+if (!process.env.PR_DIST_FILES) {
+    throw new Error('Missing or invalid "PR_DIST_FILES" env variable.');
+}
+
+/**
+ * Adapted from https://gist.github.com/zentala/1e6f72438796d74531803cc3833c039c?permalink_comment_id=4455218#gistcomment-4455218
+ * @param {number} bytes
+ * @param {number} digits
+ * @returns {string}
+ */
+function formatBytes(bytes, digits = 2) {
+    if (bytes === 0) {
+        return '0 B';
+    }
+    const sizes = [`B`, 'kB', 'MB'];
+    const i = Math.floor(Math.log(bytes) / Math.log(1024));
+
+    return parseFloat((bytes / Math.pow(1024, i)).toFixed(digits)) + ' ' + sizes[i];
+}
+
+/**
+ * @param {number} from
+ * @param {number} to
+ * @returns {number}
+ */
+function computeDiffPercent(from, to) {
+    if (from === to) {
+        return 0;
+    }
+    
+    return Number(((from - to) / to * -100).toFixed(2));
+}
+
+/**
+ * @param {number} percent
+ * @returns {string}
+ */
+function formatDiffPercent(percent) {
+    return percent > 0 ? `+${percent}% 📈` : percent < 0 ? `${percent}% 📉` : `${percent}%`;
+}
+
+export function main() {
+    let base = JSON.parse(process.env.BASE_DIST_FILES);
+    let pr = JSON.parse(process.env.PR_DIST_FILES);
+    let output = '<h1>📊 Dist files size difference</h1>\n\n';
+
+    const files = [...new Set([...Object.keys(pr), ...Object.keys(base)])].sort().reduce((acc, file) => {
+        const added = !base[file] && pr[file];
+        const removed = base[file] && !pr[file];
+        const diff_percent_size = removed ? -100 : (added ? 100 : (computeDiffPercent(base[file].size, pr[file].size)));
+        const diff_percent_size_gz = removed ? -100 : (added ? 100 : (computeDiffPercent(base[file].size_gz, pr[file].size_gz)));
+        const diff_percent_size_brotli = removed ? -100 : (added ? 100 : (computeDiffPercent(base[file].size_brotli, pr[file].size_brotli)));
+
+        if (diff_percent_size !== 0 && diff_percent_size_gz !== 0 && diff_percent_size_brotli !== 0) {
+            acc.set(file, {
+                state: added ? 'added' : (removed ? 'removed' : 'changed'),
+                diff_percent_size,
+                diff_percent_size_gz,
+                diff_percent_size_brotli
+            });
+        }
+
+        return acc;
+    }, new Map);
+
+    if (files.size === 0) {
+        output += 'ℹ️ No difference in dist files.\n';
+        return output;
+    }
+
+    output += 'Thanks for the PR! Here is the difference in size of the dist files between the base and the PR.\n';
+    output += 'Please review the changes and make sure they are expected.\n\n';
+    output += `<table>
+    <thead><tr><th>File</th><th>Diff (B)</th><th>Diff (%)</th></tr></thead>
```

Comment:
I would simplify the output to display something more similar to pck-size action, which I think it's very readable. Something like this:

```js
    <thead><tr><th>File</th><th>Before (Size / Gzip)</th><th>After (Size / Gzip)</th></tr></thead>
    <tbody>`;
    for (const [file, details] of files.entries()) {
        output += `<tr>
        <td><code>${file}</code> ${details.state === 'added' ? '(new)' : (details.state === 'removed' ? '(deleted)' : '')}</td>
        <td>
            <code>${formatBytes(base[file]?.size || 0)}</code> / <code>${formatBytes(base[file]?.size_gz || 0)}</code>
        </td>
        <td align="right">
            <sup>${formatDiffPercent(details.diff_percent_size)}</sup> (<code>${formatBytes(pr[file]?.size || 0)}</code>)
            / <sup>${formatDiffPercent(details.diff_percent_size_gz)}</sup> <code>${formatBytes(pr[file]?.size_gz || 0)}</code>)
        </td>
    </tr>
```

Reply:
This is what I've did first, but that was ugly because our filenames are very very long (but I think I can do something about it) and it was breaking the sizes into multiple lines.

I will rework that part, thanks

Reply:
I re-used your code and this is what happens:
<img width="979" alt="image" src="https://github.com/user-attachments/assets/0339ecba-b79a-4d1e-ae92-7673349e37af">

As you can see, because of line breaks the display is not very nice :/

Reply:
Also, it does not contain the brotli-compressed size of dist files, which I really want to keep (since it's more efficient than gzip and zstd)

Reply:
I reworked the table rendering to display info in a smarter way:
<img width="931" alt="image" src="https://github.com/user-attachments/assets/08c127d7-5949-4124-8bc3-4c840c466174">

You don't have useless data like `0 B` or `-+100%` displayed anymore, instead it shows you if the file has been added or removed (without displaying size) and the delta (%) only if the file has been changed.
I've regrouped files by packages, and each packages and files has a link.